packages/material-ui-based-components/src/components/Rating.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useCallback } from 'react';
+import { withRuleEngine } from '../shared/withRuleEngine';
+import { PROPS } from '../utils/types';
+import { Box, FormHelperText, InputLabel, Rating } from '@mui/material';
+
+const RatingComponent = (props: PROPS) => {
+    const { isError, name, label, required, description, enabled, readOnly, maximum, errorMessage, id, value } = props;
+
+    const changeHandler = useCallback((event: any) => {
+        props.dispatchChange(event.target.value);
+    }, [props.dispatchChange]);
+
+    const blurHandler = useCallback((event: any) => {
+        props.dispatchBlur(event.target.value);
+    }, [props.dispatchBlur]);
+
+    const focusHandler = useCallback((event: any) => {
+        props.dispatchFocus(event.target.value);
+    }, [props.dispatchFocus]);
+
+    return (
+        <Box>
```

Comment:
can we use FormControl ?

Reply:
No, on using FormControl, the layout is distorted and label overlaps over the rating component.